HW0/P5/P5.txt

```diff
@@ -1 +1,13 @@
 # Your answers here.
+1. 255 seconds. 
+2. 34 seconds.  Split into 8 equal size groups of 
+3. 8 seconds. First split into 128 groups of 2, we can add all these in 1 second. Then split the 128 sums into 64 groups of 2, which we can add in 1 second. Continuing in this fashion we are left with one sum after 8 steps, which is the sum of all of the bags. 
+
+4. log_2(N), using the above reasoning (each time we half the number of computations, we can do this log_2(N) times before we have one number, which is the total sum)
+
+5. 15 seconds. First step takes 1 second and the subsequent steps each require 1 extra second for the transfer of numbers between half of the workers. This happens for 7 steps giving a total of 7 * 2 + 1 sec nds. 
+
+6. We assume that first we must distribute all of the bags (before doing any of the additions) 
+i)  273 = 256 + 1 + 2 * 8 each intermediate step takes 2 seconds, 1 to compute the sum of two numbers, 1 to communicate the new sum to another worker. First we must pass out 256 bags. Then we need 1 second to pass bags between workers so 128 of them have 2 bags to sum. Then we run the intermediate steps of add and communicate. 
```

Comment:
It is an unfair assumption that you must distribute all the bags first (why should any particular cashier wait before every cashier receives their bags?)